test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2649,8 +2742,29 @@ def cgroup_offline_node(self, name, vnpernuma=False):
             self.skipTest('pbs_cgroups_hook: Failed to copy '
                           'freezer state FROZEN')
 
-        # Make sure the kernel knows about the freeze on MoM
-        time.sleep(1)
+        confirmed_frozen = False
+
+        for count in range(30):
+            ret = self.du.cat(hostname=self.hosts_list[0],
+                              filename=freezer_file,
+                              sudo=True)
+            if ret['rc'] != 0:
+                self.logger.info("Cannot confirm freezer state"
+                                 "sleeping 30 seconds instead")
+                confirmed_frozen = True
+                time.sleep(30)
+                break
```

Comment:
I see a "break" here, after "Cannot confirm freezer state sleeping for 30 seconds instead". Shouldn't retry checking freezer file and not have the 'break'?

Reply:
If for some reason the commands to confirm the state of the freezer don't work, the workaround is to wait 30 seconds and then bail out -- we don't know whether it's "enough" but it gives us at least some chance that the processes are frozen. The 30 seconds is the maximum we'd wait if the queries did not fail.

The problem is that we can't just skip the test when something goes wrong and be done with it. If we're flying blind and the freezer _is_ active, we can't just give up: it would poison the host for any subsequent PTL tests, until it is rebooted.

If you prefer we could time.sleep(1) and then not break, but if things are iffy I'd rather wait for as long as reasonable.

Reply:
So the way I see the code is if it can't confirm the frozen state as the 'cat freezer_file' returned a non-zero exit status, it would say confirm_frozen=true, and then sleep 30 seconds, and break, which means to not do a retry anymore of 'cat freezer_file' to see if we can get a better status. I just want to make sure that is what you intended.

Reply:
Yes, that's what I intended. Except that we should probably set confirmed_frozen to False --  we can't decide whether the test failed or succeeded.

I'll change that once you confirm you finished your (first) review.

Reply:
Update made.